isic/core/search.py

```diff
@@ -151,12 +151,15 @@ def bulk_add_to_search_index(qs: QuerySet[Image], chunk_size: int = 500) -> None
             logger.error('Failed to insert document into elasticsearch', info)
 
 
-def facets(query: Optional[dict] = None) -> dict:
+def facets(query: Optional[dict] = None, collections: Optional[list[int]] = None) -> dict:
     body = {
         'size': 0,
         'aggs': DEFAULT_SEARCH_AGGREGATES,
     }
 
+    if collections is not None:
```

Comment:
If `collections` is None, what happens? Shouldn't we still filter out collection values which the user can't access?

Reply:
Solution: `query['bool'].get('filter', {}).get('terms', {}).get('collections')` shouldn't exist. Instead, build `body['aggs']['collections']` from whatever collections the current user generally has access to see.

This is correct by analogy with any other metadata query, where even if the filter is very specific, empty buckets for all other values are still returned.